platform-sdk/platform-apps/tests/StatsSigningTestingTool/src/main/java/com/swirlds/demo/stats/signing/algorithms/ExtendedSignature.java

```diff
@@ -68,9 +70,11 @@ public ExtendedSignature(final byte[] signature) {
      * 		the raw or encoded R coordinate.
      * @param s
      * 		the raw of encoded S coordinate.
+     *
+     * @throws NullPointerException in case {@code signature} parameter is {@code null}
      */
     public ExtendedSignature(final byte[] signature, final byte[] r, final byte[] s) {
-        throwArgNull(signature, "signature");
+        Objects.requireNonNull(signature, String.format(ERROR_ARGUMENT_NULL, "signature"));
```

Comment:
I do not like that pattern. Even if `signature` is not `null` the string format will happen. That ends in a performance loss for every null check. It is boring but `"signature must not be null"` will still be the best value to pass to the method...

Reply:
Next to that `String.format` is really not what we should use when concatenating String. See the answer on this StackOverflow: https://stackoverflow.com/questions/513600/should-i-use-javas-string-format-if-performance-is-important

Adding @timo0 since he started to like `String.format`, too 😝

Reply:
yes, I see your point, agree... I only used String.format because it was the mechanism already in use in throwArgBlank and throwArgNull, but now that the check is removed, is adding an extra time penalty for positive scenarios... 
Also, I didn't like that it introduced so many lines of repeated code...
I'll change all the strings to "paramName must not be null", but sadly, it will have to be hardcoded Strings.